daml-assistant/daml-helper/src/DA/Daml/Helper/Ledger.hs

```diff
@@ -177,17 +177,28 @@ runLedgerUploadDar flags darPathM = do
       getDarPath
   putStrLn $ "Uploading " <> darPath <> " to " <> showHostAndPort args
   bytes <- BS.readFile darPath
-  uploadDarFile args bytes `catch` \(e :: SomeException) -> do
```

Comment:
Is the only way `uploadDarFile` could fail in the past through the `fail` in `either fail return` below or were the other ways? If there were other ways, we wouldn't catch these exceptions anymore.

Reply:
I changed this now to show the error caused by a `Left` return value which is caused by invalid arguments to the command, and catch exceptions and hint for the maximum upload size, since these are likely thrown by the network layer.